src/Symfony/Component/Messenger/Transport/AmqpExt/Connection.php

```diff
@@ -225,7 +225,7 @@ private function publishWithDelay(string $body, array $headers, int $delay, Amqp
     private function publishOnExchange(\AMQPExchange $exchange, string $body, string $routingKey = null, array $headers = [], AmqpStamp $amqpStamp = null)
     {
         $attributes = $amqpStamp ? $amqpStamp->getAttributes() : [];
-        $attributes['headers'] = array_merge($headers, $attributes['headers'] ?? []);
+        $attributes['headers'] = array_merge($attributes['headers'] ?? [], $headers);
```

Comment:
This looks subjective either way (or am I missing something)? The `$headers` comes ultimately from the serializer. So, should headers from the `AmqpStamp` win or headers from the serialization of the message. I dunno :). But I was curious what motivated the change.

Reply:
This just gives headers from the serialization more prio than headers from the stamp. As the headers from the stamp are copied for retry, this just makes more sense now (in case re-serialization returns different headers).

Reply:
This was huge BC break, it broke our app and make overriding `type` header impossible.